docs/coordinates/performance.inc.rst

```diff
@@ -15,6 +15,12 @@ coordinate::
 
     >>> coord = SkyCoord(ra_array, dec_array, unit='deg')  # doctest: +SKIP
 
+Frame attributes can be arrays too, as long as the coordinate data and all of
+the frame attributes have shapes that are compatible according to
+:doc:`Numpy broadcasting rules <numpy:user/basics.broadcasting>`::
+
+    >>> coord = FK4(1 * u.deg, 2 * u.deg, obstime=["J2000", "J2001"])  # doctest: +SKIP
```

Comment:
Why skip the doctest?  I'd add
```
    >>> coord.shape
    (2,)
```

Reply:
It's skipped because we haven't imported FK4. If we don't want to skip this, then we would have to add this to the doctest:

    >>> from astropy.coordinates import FK4

Reply:
Hmm, `SkyCoord` is definitely imported in `index.rst` (which includes `performance.inc.rst`, so perhaps change the example to `SkyCoord`.

Looking at the line above, there is a `skip` because `ra_array` and `dec_array` are not defined, so if you want to go a similar route, you could do here `SkyCoord(ra, dec, obstime=time_array)`

Reply:
No, that still fails:

```
============================================== FAILURES ==============================================
___________________________________ [doctest] performance.inc.rst ____________________________________
013 as opposed to creating individual |SkyCoord| objects for each individual
014 coordinate::
015 
016     >>> coord = SkyCoord(ra_array, dec_array, unit='deg')  # doctest: +SKIP
017 
018 Frame attributes can be arrays too, as long as the coordinate data and all of
019 the frame attributes have shapes that are compatible according to
020 :doc:`Numpy broadcasting rules <numpy:user/basics.broadcasting>`::
021 
022     >>> coord = SkyCoord(1 * u.deg, 2 * u.deg, obstime=["J2000", "J2001"])
UNEXPECTED EXCEPTION: NameError("name 'SkyCoord' is not defined")
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/doctest.py", line 1351, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest performance.inc.rst[1]>", line 1, in <module>
NameError: name 'SkyCoord' is not defined
/Users/lpsinger/src/astropy/docs/coordinates/performance.inc.rst:22: UnexpectedException
```

Reply:
Running `pytest docs/coordinates/index.rst` does not result in the tests in `docs/coordinates/performance.inc.rst` getting executed. I convinced myself of this by adding this to `docs/coordinates/performance.rst`, and seeing that `pytest docs/coordinates/index.rst` still passes:

```rst
Performance Tips
================

::
    >>> 1
    0
```

Reply:
I fixed this by changing it to:

```rst
Frame attributes can be arrays too, as long as the coordinate data and all of
the frame attributes have shapes that are compatible according to
:doc:`Numpy broadcasting rules <numpy:user/basics.broadcasting>`:


.. testsetup::

    >>> from astropy.coordinates import FK4
    >>> from astropy import units as u

::

    >>> coord = FK4(1 * u.deg, 2 * u.deg, obstime=["J2000", "J2001"])
    >>> coord.shape
    (2,)
```

See https://github.com/scientific-python/pytest-doctestplus#sphinx-doctest-directives.

Reply:
OK, let's just stick with what you have, or, if you are still editing the docs for @StuartLittlefair's request to show the example of `whatsnew` in the docs, change it to the more abstract `SkyCoord(ra, dec, obstime=time_array)`.

Reply:
> if you are still editing the docs for @StuartLittlefair's request to show the example of whatsnew in the docs

Right, I had forgotten about that. Fixed now.